makefile

```diff
@@ -107,16 +117,19 @@ o:	$(ALL_O)
 
 a:	$(ALL_A)
 
+%.o: %.c
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c -o $(BD)/$@ $<
+
 $(CORE_T): $(CORE_O) $(AUX_O) $(LIB_O)
-	$(AR) $@ $?
-	$(RANLIB) $@
+	$(AR) $(BD)/$@ $(addprefix $(BD)/, $?)
+	$(RANLIB) $(BD)/$@
 
 $(LUA_T): $(LUA_O) $(CORE_T)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(CORE_T) $(LIBS) $(MYLIBS) $(DL)
-
+	$(CC) -o $(BD)/$@ $(MYLDFLAGS) $(addprefix $(BD)/, $(LUA_O) $(CORE_T)) $(LIBS) $(MYLIBS) $(DL)
```

Comment:
Is changing where the outputs are done (so from running dir to cmp/XXX) really needed? This would mean also the yocto recipes need to be changed compared to usual ones and not just point to a different git repository.

Reply:
See latest comment